test_server.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+import socket
+import sys
+import pytest
+import server
+from multiprocessing import Process
+
+ADDR = ('127.0.0.1', 8000)  # port 0 may force os to find an open port
+
+STATUS200 = b"""HTTP/1.1 200 OK\r\n
+    DATE: Sun, 21 Jul 2001 23:32:15 GTM\r\n
+    SERVER: Python/2.7.6\r\n
+    \r\n"""
+
+STATUS500 = b"""HTTP 500 Internal Server Error \r\n
+    DATE: Sun, 21 Jul 2001 23:32:15 GTM\r\n
+    SERVER: Python/2.7.6\r\n
+    \r\n"""
+
+
+@pytest.yield_fixture()
+def server_process():
+    process = Process(target=server.main)
+    process.daemon = True
+    process.start()
+    yield process
+
+
+@pytest.fixture()
+def client_socket():
+
+    client_socket = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+        )
+    client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    return client_socket
+
+
+def test_start_server(server_process):
+    """Dummy function to start server"""
+    pass
+
+
+def test_response_ok():
+    assert server.response_ok() == STATUS200
```

Comment:
This is a tautology.  It's true because it's true, but it doesn't prove anything useful about the nature of the returned value of `response_ok`.  Your unit tests for that function should demonstrate that it is a proper HTTP response, not that it looks like another string, which may or may not be a proper HTTP response.